src/exoticatechnologies/integration/indevo/RelicComponentUpgradeMethod.java

```diff
@@ -67,20 +67,34 @@ public String apply(@NotNull FleetMemberAPI member, @NotNull ShipVariantAPI vari
             upgradeCost = removeCommodityAndReturnRemainingCost(storageCargo, IndEvoUtil.RELIC_COMPONENT_ITEM_ID, upgradeCost);
         }
 
-        CargoAPI fleetCargo = FleetMemberUtils.INSTANCE.findFleetForVariant(member.getVariant(), member).getCargo();
-        if (upgradeCost > 0) {
-            removeCommodity(fleetCargo, IndEvoUtil.RELIC_COMPONENT_ITEM_ID, upgradeCost);
+        //FIXME how is this one different from the other 'variant' we have and use below ???
+        ShipVariantAPI memberVariant = member.getVariant();
+        CampaignFleetAPI variantsFleet = FleetMemberUtils.INSTANCE.findFleetForVariant(memberVariant, member);
+        if (variantsFleet != null) {
+            CargoAPI fleetCargo = variantsFleet.getCargo();
+            if( upgradeCost > 0 ) {
+                removeCommodity(fleetCargo, IndEvoUtil.RELIC_COMPONENT_ITEM_ID, upgradeCost);
+            }
+
+            mods.putUpgrade(upgrade);
+            ShipModLoader.set(member, variant, mods);
+            ExoticaTechHM.addToFleetMember(member, variant);
```

Comment:
I really have no idea what to do with this, specifically the `//FIXME` comment...